luaunit.lua

```diff
@@ -1279,19 +1264,12 @@ then OK or FAILED (failures=1, error=1)
 
 ]]
 
-local TextOutput = { __class__ = 'TextOutput' }
-local TextOutput_MT = { -- class
-    __index = TextOutput
-}
+local TextOutput = { __class__ = 'TextOutput' } -- class
+local TextOutput_MT = { __index = TextOutput } -- metatable
 
     function TextOutput:new()
-        local t = {}
-        t.runner = nil
-        t.result = nil
-        t.errorList ={}
-        t.verbosity = M.VERBOSITY_DEFAULT
-        setmetatable( t, TextOutput_MT )
-        return t
+        return setmetatable(
```

Comment:
There same goes for other private fields that get initialized from within `LuaUnit.startSuite()` (https://github.com/bluebird75/luaunit/blob/b5fd780ec59be20c3664809094f8b4d1583549bc/luaunit.lua#L1735-L1738), namely `verbosity` and `fname`.

Setting them "across" classes like that is probably a bad idea anyway. Maybe we should include those options explicitly in the constructor - something like `self.output = self.outputType:new(self, self.result, self.verbosity, self.fname)`.

Unfortunately, implementing it like that currently opens just another can of worms. There are unit tests that rely on the side effects (assigning the outputter's fields) of `LuaUnit.startSuite()` - using a custom outputter like in https://github.com/bluebird75/luaunit/blob/b5fd780ec59be20c3664809094f8b4d1583549bc/test/test_luaunit.lua#L2475-L2476 therefore tends to break easily...